woocommerce/woocommerce
# Add date index to order tables for speeding up order list query.

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This adds the equivalent index of wp_posts' `type_status_date`, the main impact of this index would be on the order list page for large sites. Additionally, it also connects WC_Install's create table method to HPOS when it is enabled.

Closes #35465 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Set HPOS to authoritative, or data sync to enabled, or both. (So that HPOS tables are registered and created).
2. Testing this needs to trigger the update flow, the easiest option is to update the WooCommerce version to a lower one, like so: 
`wp option update woocommerce_version 7.9.0`
3. Refresh any WooCommerce admin page, so that update is triggered. 
4. Verify that woocommerce version was updated, by running `wp option get woocommerce_version`, it should be greater than 7.9.0.
5. Check that the index was created as expected, by running ` wp db query "show index from wp_wc_orders where Key_name like 'type_status_date'"`, it should show 3 rows for each index in query like this:
```
+--------------+------------+------------------+--------------+------------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| Table        | Non_unique | Key_name         | Seq_in_index | Column_name      | Collation | Cardinality | Sub_part | Packed | Null | Index_type | Comment | Index_comment |
+--------------+------------+------------------+--------------+------------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
| wp_wc_orders |          1 | type_status_date |            1 | type             | A         |           3 |     NULL | NULL   | YES  | BTREE      |         |               |
| wp_wc_orders |          1 | type_status_date |            2 | status           | A         |          10 |     NULL | NULL   | YES  | BTREE      |         |               |
| wp_wc_orders |          1 | type_status_date |            3 | date_created_gmt | A         |         309 |     NULL | NULL   | YES  | BTREE      |         |               |
+--------------+------------+------------------+--------------+------------------+-----------+-------------+----------+--------+------+------------+---------+---------------+
```



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
